dds/DCPS/Service_Participant.h

```diff
@@ -761,6 +764,9 @@ class OpenDDS_Dcps_Export Service_Participant {
   /// Used to track state of service participant
   bool shut_down_;
 
+  /// Protects the internal shutdown state tracking
+  mutable ACE_Thread_Mutex shut_down_lock_;
+
```

Comment:
Why not make the shutdown flag an atomic?

Reply:
Atomic flags are great, until you need two of them to represent your state. Without knowing more about what state is being tracked in Service_Participant, this seemed like the safer option. If we're confident it's just the one flag and it doesn't need to be coordinated with any other data structures, sure, I can switch to an atomic.